README.md: updates (#42)
connylabs/ingest: README.md

# ingest

> Ingest is a pluggable tool that makes it easy to orchestrate the synchronize of objects across multiple sources.

Reviewer: ```suggestion
Ingest is a pluggable tool that makes it easy to orchestrate the synchronization of objects from any API into an object storage system
```

## Concept

> Ingest includes a main binary and some common plugins like s3.

Reviewer: ```suggestion
Ingest includes a main binary and some plugins for some common storage systems, like S3.
```

> Each plugin can implement a data source, target or both.

Reviewer: ```suggestion
Each plugin can implement a data source, a target or both.
```

The S3 plugin implements both the source and target interface, but some plugins may only be able to act as either of them.
The plugins are loaded at runtime and enable users to implement their own custom plugins.

### Workflows

> A workflow defines a data source and one or more targets.

Reviewer: ```suggestion
A workflow specifies a data source and one or more targets.
```

When configured, objects from the source will be copied to all targets.

## Configuration

The exact configurations depends on the plugin.
Typically, the S3 plugin requires an URL, an access key and a secret access key.
This may differ from plugins that don't require authentication or a different kind of authentication.

The following example shows the configuration to copy object between two instances of S3:

```yaml
sources:
- name: foo_1
  type: s3
  endpoint: source.amazon.com
  bucket: source
  prefix: prefix/
  accessKeyID: key
  secretAccessKey: secret
destinations:
- name: bar_1
  type: s3
  endpoint: destination.amazon.com
  insecure: true
  bucket: destination
  prefix: prefix1/
  metafilesPrefix: meta/
  accessKeyID: key
  secretAccessKey: secret
workflows:
- name: foo_1-bar_1
  source: foo_1
  destinations:
  - bar_1
  batchSize: 1
  interval: 300s
  cleanUp: true
  webhook: http://localhost:8080
```

## Deployment

The deployment of ingest contains of two parts.

> On part is called the enqueuer.

Reviewer: ```suggestion
One part is called the enqueuer.
```
